yarn-project/merkle-tree/src/snapshots/append_only_snapshot.ts

```diff
@@ -0,0 +1,179 @@
+import { Hasher, SiblingPath } from '@aztec/types';
+
+import { LevelUp } from 'levelup';
+
+import { AppendOnlyTree } from '../interfaces/append_only_tree.js';
+import { SiblingPathSource } from '../interfaces/merkle_tree.js';
+import { TreeBase } from '../tree_base.js';
+import { SnapshotBuilder } from './snapshot_builder.js';
+
+const nodeVersionKey = (name: string, level: number, index: bigint) =>
+  `snapshot:${name}:node:${level}:${index}:version`;
+const nodePreviousValueKey = (name: string, level: number, index: bigint) =>
```

Comment:
I don't think this one is actually `previous`, consider changing to just `nodeValueKey`.